src/main.rs

```diff
@@ -164,5 +175,46 @@ fn main() {
 
             CliResult::new(result, FailureExitCode::Upload).print_or_exit(format);
         }
+        Commands::DeleteCollection {
+            giant_uri,
+            collection,
+        } => {
+            let result: Result<(), CliError> = (|| {
+                // Returns a maximum of 500 results,
+                // so we need to loop until we've deleted them all.
+                let mut blobs = giant_api::get_blobs_in_collection(giant_uri, collection)?;
+
+                while !blobs.is_empty() {
+                    for blob in blobs {
+                        println!("Blob is in collections: {:?}", blob.collections);
+
+                        let other_collections: Vec<String> = blob
+                            .collections
+                            .into_iter()
+                            .filter(|c| c != collection)
+                            .collect();
+
+                        if !other_collections.is_empty() {
+                            println!(
+                                "Blob {} exists in other collections, will also delete from: {:?}",
+                                blob.uri, other_collections
+                            );
+                        }
+                        println!("Deleting blob {}", blob.uri);
+                        giant_api::delete_blob(giant_uri, &blob.uri)?;
+                        println!("Deleted blob {}", blob.uri);
+                    }
+                    blobs = giant_api::get_blobs_in_collection(giant_uri, collection)?;
+                }
+
+                println!("Deleting collection {}", collection);
+                giant_api::delete_collection(giant_uri, collection)?;
+                println!("Deleted collection {}", collection);
+
+                return Ok(());
```

Comment:
Slightly unidiomatic to have a return statement at the end of a block.

You can just have the last line be the expression `Ok(())` and it will be return value.

Reply:
Note, you need to not include the `;` because that makes the expression into a statement. (Sorry if you already know this)